tests/unit/test_routes.py

```diff
@@ -371,6 +371,66 @@ def test_update_resource(module_client, module_db, fake_auth_from_oc):
     assert (response.status_code == 404)
 
 
+def test_search(module_client, module_db, fake_auth_from_oc):
+    client = module_client
+    first_term = names.get_full_name()
+    apikey = get_api_key(client)
+
+    # Create resource and find it in the search results.
+    resource = client.post("/api/v1/resources",
+                          json=dict(
+                              name=f"{first_term}",
+                              category="Website",
+                              url=f"{first_term}",
+                              paid=False,
+                          ),
+                          headers={'x-apikey': apikey}
+                          )
+    result = client.get(f"/api/v1/search?q={first_term}")
+
+    # Prevents false negative from race condition.
+    while result.status_code == 302:
+        time.sleep(.5)
+        result = client.get(f"/api/v1/search?q={first_term}")
+
+    assert (resource.status_code == 200)
+    assert (result.status_code == 200)
+
+    found = False
+    for entry in result.json['data']:
+        if entry.get('url') == resource.json['data'].get('url'):
+            found = True
```

Comment:
I feel like there's a more pythonic way to do these checks. Once we get the mocking down, I'm going to play with this and see if I can come up with a bit of a better way to see if the resource's URL is in the search results. Maybe this is the best way to do it, but maybe not

Reply:
After mocking the search, this is pretty irrelevant since we will only return the results we want. We could just do `assert (results['data'][0]['url'] == resource.json['data'].get('url')` for O(1) time if you were concerned with iterating.